src/rust/ide/enso-protocol/src/language_server/types.rs

```diff
@@ -37,6 +37,14 @@ impl Display for Path {
 }
 
 impl Path {
+    /// Splits path into name and segments. e.g.:
+    /// Path{root_id,segments:["foo","bar","qux"]} => ("qux",Path{root_id,segments:["foo","bar"]})
+    pub fn split_into_name_and_segments(&self) -> (String,Path) {
+        let mut path = self.clone();
+        let name     = path.segments.pop().unwrap_or_default();
```

Comment:
I'd rather not unwrap but return return Option, for consistency with `file_name`.
If there are no segments in the path, splitting it into two pieces makes no sense.

Reply:
Also, for consistency, like we have `file_name` we could also introduce `parent` and then implement this function in terms of these two.